src/Symfony/Bridge/Twig/ErrorRenderer/TwigErrorRenderer.php

```diff
@@ -46,7 +47,9 @@ public function render(\Throwable $exception): FlattenException
         $debug = \is_bool($this->debug) ? $this->debug : ($this->debug)($flattenException);
 
         if ($debug || !$template = $this->findTemplate($flattenException->getStatusCode())) {
-            return $this->fallbackErrorRenderer->render($exception);
+            $renderer = \in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) ? new CliErrorRenderer() : $this->fallbackErrorRenderer;
```

Comment:
wouldn't this break cases using our ErrorRenderer component with some CLI-based application server ?

Couldn't we do this SAPI check in the constructor instead, when choosing the default `fallbackErrorRenderer`, so that such project can still force using the HtmlErrorRenderer by passing one explicitly ?

Reply:
Agree, in the constructor probably better to have something like:
```php
$this->fallbackErrorRenderer = $fallbackErrorRenderer ?? (\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) ? new CliErrorRenderer() : new HtmlErrorRenderer());
```